src/js/gi/selectors/calculator.js

```diff
@@ -0,0 +1,23 @@
+import { createSelector } from 'reselect';
+
+const getConstants = (state) => state.constants.constants;
+
+const getEligibilityDetails = (state) => {
+  const details = Object.assign({}, state.eligibility);
+  delete details.dropdowns;
```

Comment:
The use of reselect looks fine to me. Because some of the attribute calculations are fairly complicated, reselect lets us only recalculate when needed.